src/nvim/ex_docmd.c

```diff
@@ -5600,6 +5600,42 @@ static void ex_detach(exarg_T *eap)
   }
 }
 
+/// ":connect"
+///
+/// Connects the current UI to a different server
+///
+/// ":connect <address>" connects the currently running TUI to the server at the
+/// given address and detaches the connected server
+/// ":connect! <address>" (with the bang (!)) connects the currently running TUI
+/// to the server at the given address and stops the connected server if no
+/// other UIs are attached to it
+static void ex_connect(exarg_T *eap)
+{
+  if (eap->argc > 1) {
+    emsg("connect: only one argument is allowed");
+    return;
+  }
+
+  bool stop_server = false;
+  if (eap->forceit) {
+    stop_server = (get_ui_number() == 1);
+  }
+
+  Error err = ERROR_INIT;
+  remote_ui_connect(current_ui, eap->arg, &err);
+  if (ERROR_SET(&err)) {
+    emsg(err.msg);
+    api_clear_error(&err);
+  }
+
+  if (stop_server) {
+    exiting = true;
+    getout(0);
+  } else {
+    ex_detach(NULL);
```

Comment:
detach should probably happen before `remote_ui_connect` ? seems wrong to emit a "connect" event followed by "detach".

Reply:
How can a UI receive a `connect` event when it's already detached?